lib/ansible/modules/identity/cyberark/cyberark_authentication.py

```diff
@@ -0,0 +1,304 @@
+
+            if state == "present":
+
+                if useSharedLogonAuthentication:
+                    token = json.loads(response.read())["LogonResult"]
+                else:
+                    token = json.loads(response.read())["CyberArkLogonResult"]
+
+                result = {
+                    "cyberarkSession": {
+                        "token": token,
+                        "apiBaseUrl": apiBaseUrl,
+                        "validate_certs": validate_certs,
+                        "useSharedLogonAuthentication":
+                        useSharedLogonAuthentication},
+                }
+
+                if newPassword is not None:
+                    changed = True
+
+            else:
+                result = {
+                    "cyberarkSession": {}
+                }
+
+            return (changed, result, response.getcode())
+        else:
+            module.fail_json(
+                msg="error in endpoint=>" +
+                endPoint +
+                json.dumps(headers))
+
+    except httplib.HTTPException:
+        t, e = sys.exc_info()[:2]
+        module.fail_json(
+            msg="EndPoint=" +
+            apiBaseUrl +
+            endPoint +
+            " headers=[" +
+            json.dumps(headers) +
+            "] ===>" +
+            str(e),
+            exception=traceback.format_exc(),
+            status_code=e.code)
+    except Exception:
+        t, e = sys.exc_info()[:2]
+        module.fail_json(
+            msg="EndPoint=" +
+            apiBaseUrl +
+            endPoint +
+            " headers=[" +
+            json.dumps(headers) +
+            "] ===>" +
+            str(e),
+            exception=traceback.format_exc(),
+            status_code=-1)
+
+
+def main():
+
+    fields = {
+        "apiBaseUrl": {"required": False, "type": "str"},
+        "validate_certs": {"required": False, "type": "bool",
+                          "default": "true"},
+        "username": {"required": False, "type": "str"},
+        "password": {"required": False, "type": "str", "no_log": True},
+        "newPassword": {"required": False, "type": "str", "no_log": True},
+        "useSharedLogonAuthentication": {"default": False, "type": "bool"},
+        "useRadiusAuthentication": {"default": False, "type": "bool"},
+        "state": {"required": False, "type": "str",
+                  "choices": ["present", "absent"],
+                  "default": "present"},
+        "cyberarkSession": {"required": False, "type": "dict"},
+    }
+
+    required_if = [
+        ("state", "present", ["apiBaseUrl"]),
+        ("state", "absent", ["cyberarkSession"])
+    ]
+
+    required_together = [
+        ["username", "password"]
+    ]
+
+    module = AnsibleModule(
+        argument_spec=fields,
+        required_if=required_if,
+        required_together=required_together,
+        supports_check_mode=True)
+
+    if module.check_mode:
+        module.exit_json(change=False)
```

Comment:
@bjolivot thanks for the suggestion, I changed the logic internally that when running in check_mode it can be used to test authentication, and not actually performing password changes when passing a new_password.